internal/provider/resource_ko_image.go

```diff
@@ -124,6 +130,18 @@ type buildOptions struct {
 	sbom       string
 }
 
+var (
+	amazonKeychain authn.Keychain = authn.NewKeychainFromHelper(ecr.NewECRHelper(ecr.WithLogger(ioutil.Discard)))
+	azureKeychain  authn.Keychain = authn.NewKeychainFromHelper(credhelper.NewACRCredentialsHelper())
+	keychain                      = authn.NewMultiKeychain(
+		amazonKeychain,
+		authn.DefaultKeychain,
```

Comment:
Order matters, so despite alphabetical being nice, I think the default keychain should trump in terms of precedence.